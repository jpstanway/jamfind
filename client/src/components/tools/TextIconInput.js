import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";

const TextIconInput = ({
  label,
  icon,
  type,
  name,
  placeholder,
  value,
  onChange,
  disabled,
  error,
  info
}) => {
  return (
    <div className="input-group mb-3">
      {label && <label htmlFor={name}>{label}</label>}
      <div className="input-group-prepend">
        <span className="input-group-text" id={name}>
          <i className={icon} />
        </span>
      </div>
      <input
        type={type}
        className={classnames("form-control", {
          "is-invalid": error
        })}
        name={name}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        disabled={disabled}
      />
      {info && <small className="form-text text-muted">{info}</small>}
      {error && <div className="invalid-feedback">{error}</div>}
    </div>
  );
};

TextIconInput.defaultProps = {
  type: "text"
};

TextIconInput.propTypes = {
  name: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.string,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  error: PropTypes.string,
  info: PropTypes.string
};

export default TextIconInput;
