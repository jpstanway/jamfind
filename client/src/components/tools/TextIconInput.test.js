import React from "react";
import { shallow } from "enzyme";
import TextIconInput from "./TextIconInput";

describe("<TextIconInput />", () => {
  let mockOnChangeHandler, mockInputData;

  beforeEach(() => {
    mockOnChangeHandler = jest.fn();
    mockInputData = {
      label: "test",
      icon: "",
      type: "text",
      name: "test",
      placeholder: "type something",
      value: "",
      onChange: mockOnChangeHandler,
      disabled: "false",
      error: "test",
      info: "test",
    };
  });

  it("renders icon input", () => {
    expect(shallow(<TextIconInput {...mockInputData} />)).toMatchSnapshot();
  });

  it("can change value of input", () => {
    const wrapper = shallow(<TextIconInput {...mockInputData} />);
    wrapper
      .find("input")
      .simulate("change", { target: { name: "test", value: "a" } });
    expect(mockOnChangeHandler.mock.calls.length).toBe(1);
  });
});
