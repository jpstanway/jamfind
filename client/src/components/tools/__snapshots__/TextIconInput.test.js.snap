// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextIconInput /> renders icon input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextIconInput
    disabled="false"
    error="test"
    icon=""
    info="test"
    label="test"
    name="test"
    onChange={[MockFunction]}
    placeholder="type something"
    type="text"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
        >
          test
        </label>,
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="test"
          >
            <i
              className=""
            />
          </span>
        </div>,
        <input
          className="form-control is-invalid"
          disabled="false"
          name="test"
          onChange={[MockFunction]}
          placeholder="type something"
          type="text"
          value=""
        />,
        <small
          className="form-text text-muted"
        >
          test
        </small>,
        <div
          className="invalid-feedback"
        >
          test
        </div>,
      ],
      "className": "input-group mb-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="input-group-text"
            id="test"
          >
            <i
              className=""
            />
          </span>,
          "className": "input-group-prepend",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className=""
            />,
            "className": "input-group-text",
            "id": "test",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control is-invalid",
          "disabled": "false",
          "name": "test",
          "onChange": [MockFunction],
          "placeholder": "type something",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "form-text text-muted",
        },
        "ref": null,
        "rendered": "test",
        "type": "small",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "invalid-feedback",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
          >
            test
          </label>,
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="test"
            >
              <i
                className=""
              />
            </span>
          </div>,
          <input
            className="form-control is-invalid"
            disabled="false"
            name="test"
            onChange={[MockFunction]}
            placeholder="type something"
            type="text"
            value=""
          />,
          <small
            className="form-text text-muted"
          >
            test
          </small>,
          <div
            className="invalid-feedback"
          >
            test
          </div>,
        ],
        "className": "input-group mb-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="input-group-text"
              id="test"
            >
              <i
                className=""
              />
            </span>,
            "className": "input-group-prepend",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className=""
              />,
              "className": "input-group-text",
              "id": "test",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control is-invalid",
            "disabled": "false",
            "name": "test",
            "onChange": [MockFunction],
            "placeholder": "type something",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "form-text text-muted",
          },
          "ref": null,
          "rendered": "test",
          "type": "small",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "invalid-feedback",
          },
          "ref": null,
          "rendered": "test",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
